### ISSUE TITLE
Use '+' on git prompt when there are uncommitted changes in the index

### ISSUE THREAD

**Issue by gauteh** [sindresorhus/pure#364]
It would be nice with a '+' after having added files using git add to the index, rather than just still having the `'*` around. If there are both a dirty worktree and uncommitted files it should both be `+*` as per the usual bash git prompt.

**Comment by gauteh** [comment]
Non-async code is essentialy already available in git-prompt.sh: https://github.com/git/git/blob/master/contrib/completion/git-prompt.sh#L483 (I don't know if any other functionality is missing from that as well, in which case I'd like to see that as well!)

**Comment by mafredri** [comment]
Could you motivate what value this suggestion bring to your workflow?

My opinion is that the `+`-sign would add visual noise without very much value. It's not like the `+` can tell you anything actionable. 

Let's say you see the `+` and do a commit & push, just to realize a moment later you hadn't added file `foo.bar` because you didn't check `git status` beforehand. This might be a contrived example, but should illustrate my point.

**Comment by gauteh** [comment]
If you make some changes, add them, and then keep making some minor changes before committing the `*+` will immediately tell you that you have more changes that you can add - without having to do an extra git status. I use this quite a lot, to the point that just seeing the `*` now usually means that I have missed something when I am typing `git commit`.

**Comment by mafredri** [comment]
Even if you stage your changes, Pure shows `*` until you've committed, so showing `*+` would not tell you that you have new unstaged changes.

**Comment by gauteh** [comment]
What do you mean? That is my suggestion; that it should show a `*` only if there are unstaged changes - as per the behavior of the official git-bash prompt. (please excuse the confusing title..)

**Comment by mafredri** [comment]
Ah, I understand what you mean now. I've never used that git prompt so I'm not familiar with how it behaves.

However, making this change would break current Pure behavior, so it can't be made lightly. I'll try to experiment with this a bit in the future and try to figure out whether or not it's a good fit for Pure.

**Comment by gauteh** [comment]
Ok, thanks, I've found this feature to be very useful! There are also symbols for stashed changes etc..

**Comment by mafredri** [comment]
Git stash has been discussed previously in #317 and #365. Let's keep that as a separate discussion. We want Pure to stay pure, and that involves keeping visual noise to a minimum, so each feature must be weighed against that.

**Comment by balki** [comment]
I would like to see distinction between staged/unstaged/untracked files. All the three is just shown as '*' now. May be just a color change for '*' is good enough.

**Comment by igoradamenko** [comment]
Powerlevel10k has support for untracked files, and it looks like this:

```
~/repo master
❯ touch file

~/repo master?
❯ git add .

~/repo master+
❯

~/repo master+
❯ touch file2

~/repo master+?
❯
```

When files are untracked, it shows `?`. When they're staged it shows `+`. It shows both when there're untracked and staged.

It would be nice to have the same feature in Pure.

**Comment by welcome-linja** [comment]
Pitching in to say this would be really really nice and I'm considering switching to p10k if it isn't implemented.

I do see the point about visual noise. I propose a mitigation: if there are _any_ untracked files, it shows `?`; if all files are tracked or ignored but there are _any_ unstaged changes, it shows `*`; and only if _all_ changes are staged will it show `+`. That way, it's always at most one character, yet the _most pertinent detail_ is always displayed. @mafredri, what do you think?
